firedrake/mesh.py

```diff
@@ -2254,7 +2249,7 @@ def compare_arrays(x, y, datatype):
     return vmesh
 
 
-def _pic_swarm_in_plex(plex, coords, fields=[]):
+def _pic_swarm_in_plex(parent_mesh, coords, fields=None):
```

Comment:
By changing the input argument from `plex` to `parent_mesh` you have slightly muddied the functionality of this function I think. Are all meshes have guaranteed to have a DMPlex? The VertexOnlyMesh certainly doesn't. I preferred passing the plex in directly

Reply:
I think rather than fiddling with this function you can just call
```
    coords, reference_coords, parent_cell_nums = \
        _parent_mesh_embedding(coords, parent_mesh)
```
after 
```
swarm = _pic_swarm_in_plex(mesh.topology.topology_dm, vertexcoords, fields=[("parentcellnum", 1, IntType), ("refcoord", tdim, RealType)])
```
and then fiddle about with replacing the field entries. 

As far as I can tell we aren't getting any performance benefit from doing it inside this function since the default PETSc embedding is still going on. This would also keep the diff nice and tidy.

Reply:
Having discussed, let's just rename this `_pic_swarm_in_mesh`

Reply:
Done.